### PR TITLE
Onboarding flow frontend

### PR DESCRIPTION
**Pull request by rkerins91**
Just some notes, the "AddExperienceButton.js" is mostly a placeholder for now, will probably end up swapping the button out with some material ui component when I get to styling, as well as delete button. I also plan on adding functionality to only allow users to select one of each language, because as of now they can keep adding the same language as many times as they want.

Update: Added keys to lists, created utils file to store languages array, added functionality to only allow user to select a language once, API call is working. 

Update: 2/15: Connected to user context, dynamically sets api call user id, changed redirect after signup to experience component

TO-DO: 
- Auto-populate initial state from DB to show languages user already has knowledge of
- Styling

### CODE REVIEW

File: client/src/pages/Experience/Experience.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+const Experience = props => {
+  const [experience, setExperience] = useState([{ C: 1 }]);
+
+  const languagesAvailable = [
+    "C",
+    "C++",
+    "Java",
+    "JavaScript",
+    "Python",
+    "Ruby"
```
Comment (skassam21) [inline client/src/pages/Experience/Experience.js]: can we put this in a utils folder or something... imagine we will use it throughout

File: client/src/pages/Experience/Experience.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  const deleteExperience = idx => {
+    let newExperienceList = experience.slice();
+    newExperienceList.splice(idx, 1);
+    setExperience(newExperienceList);
+  };
+
+  return (
+    <SignUpContainer>
+      <p>Add your experience here</p>
```
Comment (skassam21) [inline client/src/pages/Experience/Experience.js]: Use material-ui components - like Typography, Button, etc.

File: client/src/pages/Experience/Experience.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      {experience.map((_, idx) => (
+        <NewExperienceForm
```
Comment (skassam21) [inline client/src/pages/Experience/Experience.js]: You need a key for this (you probably getting a warning)

File: client/src/pages/Experience/NewExperienceForm.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        labelId="label"
+        id="select"
+        onChange={handleChange("level")}
+        value={selected.level}
+        name="level"
+      >
+        <MenuItem value="1">Beginner</MenuItem>
+        <MenuItem value="2">Intermediate</MenuItem>
+        <MenuItem value="3">Advanced</MenuItem>
+        <MenuItem value="4">Expert</MenuItem>
```
Comment (skassam21) [inline client/src/pages/Experience/NewExperienceForm.js]: make this a list in a utils as well and map through it

**Comment by skassam21** [comment]
The title of this PR btw is not relevant, probably want to make it more descriptive

File: client/src/pages/Experience/Experience.js [hunk 5]
```diff
@@ -2,26 +2,39 @@ import React, { useState, useEffect } from "react";
 import SignUpContainer from "../../components/SignUpContainer";
 import NewExperienceForm from "./NewExperienceForm";
 import AddExperienceButton from "./AddExperienceButton";
+import { availableLanguages } from "../../utils";
```
Comment (skassam21) [inline client/src/pages/Experience/Experience.js]: can use absolute paths for the src root in these cases (I rarely like seeing ../, makes it harder in the future)

File: client/src/pages/Experience/Experience.js [hunk 6]
```diff
@@ -2,26 +2,39 @@ import React, { useState, useEffect } from "react";
...
 
-// TO-DO: ADD FUNCTIONALITY TO FILTER LANGUAGES ALREADY SELECTED
-const Experience = props => {
+const Experience = () => {
   const [experience, setExperience] = useState([{ C: 1 }]);
```
Comment (skassam21) [inline client/src/pages/Experience/Experience.js]: Lol that's a funny default

Comment (skassam21) [reply client/src/pages/Experience/Experience.js]: I feel like a better and more efficient structure of experience is as an object, like this:
{
"C": 1
}
It'll be easier to update

File: client/src/pages/Experience/AddExperienceButton.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const AddExperienceButton = ({ addExperience }) => {
+  return <button onClick={addExperience}> Add Another Experience </button>;
```
Comment (ashan36) [inline client/src/pages/Experience/AddExperienceButton.js]: Going to want to swap this for a material ui button for consistency

File: client/src/pages/Experience/Experience.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    // for unselected languages, preventing nested looping by having to search through
+    // 'experience' state on each pass of filter
+    const knownLanguages = {};
+    experience.forEach(ele => {
+      knownLanguages[Object.keys(ele)[0]] = true;
+    });
+
+    // Make array of languages from copy of availableLanguages array if they are
+    // not already selected
+    const newUnselectedLanguages = availableLanguages.slice().filter(ele => {
```
Comment (ashan36) [inline client/src/pages/Experience/Experience.js]: Does slice() have a function here? Array.filter() doesn't mutate the original array, it already returns a new filtered array.

File: client/src/pages/Experience/NewExperienceForm.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  const handleChange = name => async evt => {
+    setSelected({ ...selected, [name]: evt.target.value });
+  };
+
+  useEffect(() => {
+    updateExperience(index, { [selected.language]: Number(selected.level) });
+  }, [index, selected, updateExperience]);
+
+  return (
+    <>
```
Comment (ashan36) [inline client/src/pages/Experience/NewExperienceForm.js]: Probably meant to delete this

**Pull request merged**